### PR TITLE
🌱 Bump dependencies: CAPI v1.2.2 , BMO v0.1.1, IPAM v1.2.0 and downgrade k8s modules to v0.24.2 to match CAPI release-1.2

### PR DESCRIPTION
**Pull request by kashifest**

### CODE REVIEW

**Comment by kashifest** [comment]
/hold
Until new IPAM and BMO releases are out.

**Comment by kashifest** [comment]
Forgot to push make generate changes, I shall do that

File: go.mod [hunk 1]
```diff
@@ -89,7 +87,7 @@ require (
 	github.com/mailru/easyjson v0.7.6 // indirect
 	github.com/mattn/go-isatty v0.0.14 // indirect
 	github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 // indirect
-	github.com/metal3-io/baremetal-operator/pkg/hardwareutils v0.0.0 // indirect
+	github.com/metal3-io/baremetal-operator/pkg/hardwareutils v0.1.0 // indirect
```
Comment (furkatgofurov7) [inline go.mod:90]: `v0.1.1`?

Comment (kashifest) [reply go.mod]: Good catch. I failed to understand why it was not automatic.

**Comment by kashifest** [comment]
> Forgot to push make generate changes, I shall do that

Done

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by smoshiur1237 — COMMENTED** [review]
LGTM

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**